webapp/src/js/controllers/reports.js

```diff
@@ -49,6 +51,34 @@ angular
 
     var lineage = lineageFactory();
 
+    // Render the facilities hierarchy as the user is scrolling through the list
+    // Initially, don't load/render any
+    $scope.totalFacilitiesDisplayed = 0;
+    $scope.facilities = [];
+
+    // Load the facilities hierarchy and render one district hospital 
+    // when the user clicks on the filter dropdown
+    $scope.monitorFacilityDropdown = () => {
+      PlaceHierarchy()
+        .then(function(hierarchy) {
+          $scope.facilities = hierarchy;
+          $scope.totalFacilitiesDisplayed += 1;
+        })
+        .catch(function(err) {
+          $log.error('Error loading facilities', err);
+        });
+
+      $document[0].querySelector('#facilityDropdown span.dropdown-menu > ul')
+                  .addEventListener('scroll', (event) => {
+        // visible height + pixel scrolled >= total height - 100
+        if (event.target.offsetHeight + event.target.scrollTop >= event.target.scrollHeight - 100) {
+          $scope.$apply(function() {
```

Comment:
Instead of $scope.$apply I recommend using $timeout - it does the same thing (tells angular to update the DOM) but doesn't risk erroring if the digest is already in progress. eg:

```js
$timeout(() => $scope.totalFacilitiesDisplayed++);
```